src/app/api/[[...route]]/routes/sharehouse.route.ts

```diff
@@ -172,9 +181,193 @@ app.delete('/:shareHouseId', async (c) => {
   }
 });
 
-app.post('/:landlordId', (c) => {
-  const landlordId = c.req.param('landlordId');
-  return c.json({
-    message: `Landlord id for creating new share hose: ${landlordId}`,
-  });
+app.post('/', zValidator('json', shareHouseCreationSchema), async (c) => {
+  try {
+    const session = await auth();
+
+    if (!session) {
+      return c.json({
+        message: 'You are not logged in!',
+      });
+    }
+    const landlordId = session.user.id;
+    const data = c.req.valid('json');
+
+    const landlord = await prisma.landlord.findUnique({
+      where: {
+        id: landlordId,
+      },
+      include: {
+        shareHouses: true,
+      },
+    });
+
+    if (!landlord)
+      return c.json(
+        { error: 'Landlord not found for the given landlordId' },
+        404,
+      );
+
+    if (!session.user.email)
+      return c.json({ error: 'User email is missing' }, 400);
+
+    // Check for existing shareHouse by name
+    const existingShareHouse = await prisma.shareHouse.findFirst({
+      where: {
+        name: data.name,
+      },
+    });
+
+    if (existingShareHouse)
+      return c.json({ error: 'ShareHouse name already exists' }, 400);
+
+    // Check for existing categories by name
+    const existingCategories = await prisma.category.findMany({
+      where: {
+        name: {
+          in: data.categories.map((category) => category.category),
+        },
+      },
+    });
+
+    if (existingCategories.length > 0)
+      return c.json({ error: 'Category name(s) already exists' }, 400);
+
+    // Check for existing tenants by email
+    const existingTenants = await prisma.tenant.findMany({
+      where: {
+        email: {
+          in: data.tenants.map((tenant) => tenant.email),
+        },
+      },
+    });
+
+    if (existingTenants.length > 0)
+      return c.json({ error: 'Tenant email(s) already exists' }, 400);
```

Comment:
Could you remove the existing tenant email check? Since this is creating a brand new sharehouse, the tenant emails can be the same if the sharehouses are different.